pkgs/development/python-modules/scholarly/default.nix

```diff
@@ -0,0 +1,55 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, unittestCheckHook
+
+, arrow
+, beautifulsoup4
+, bibtexparser
+, deprecated
+, fake-useragent
+, free-proxy
+, httpx
+, python-dotenv
+, selenium
+, sphinx-rtd-theme
+, typing-extensions
+}:
+
+buildPythonPackage rec {
+  pname = "scholarly";
+  version = "1.7.11";
+
+  # Pypi does not contain tests
+  src = fetchFromGitHub {
+    owner = "scholarly-python-package";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = lib.fakeHash;
+  };
+
+  propagatedBuildInputs = [
+    arrow
+    beautifulsoup4
+    bibtexparser
+    deprecated
+    fake-useragent
+    free-proxy
+    httpx
+    python-dotenv
+    selenium
+    sphinx-rtd-theme
+    typing-extensions
+  ];
+
+  checkPhase = ''
```

Comment:
Add the hooks `runHook preCheck` and `runHook postCheck` here, with the test run in between.